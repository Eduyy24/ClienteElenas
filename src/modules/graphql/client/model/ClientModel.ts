/**
 * Define el modelo para un Ciudad
 * @class
 */
export class CityModel {
  id = '';
  name = '';
}

/**
 * Define el modelo para un Estado
 * @class
 */
export class StateModel {
  id = '';
  name = '';
  shortCode = '';
  cities = Array<CityModel>(0);
}

/**
 * Define el modelo para address
 * @class
 */
 export class AddressModel {
  streetAddress = ''
  city = ''
  cityId = 0
  stateShortCode = ''
  stateId = 0
  country = ''
}


/**
 * Define el modelo de salida para un cliente
 * @class
 */
export class ClientOutputModel {
  id = '';
  firstName = '';
  lastName = '';
  cedula = '';
  address = '';
  cellphone = '';
  email = '';
  city = '';
  state?: StateModel;
}


/**
 * Define el modelo de entrada para un cliente
 * @class
 */
 export  class ClientInputModel {
  firstName = '';
  lastName = '';
  cedula = '';
  address = new AddressModel();
  cellphone = '';
  email = '';
}