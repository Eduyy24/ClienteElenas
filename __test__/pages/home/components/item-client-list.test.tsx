import React from 'react';
import { fireEvent, render } from '@testing-library/react-native'
import ItemClientList from '../../../../src/pages/home/components/item-client-list';
import { ClientOutputModel } from '../../../../src/modules/graphql/client/model/ClientModel';


describe('<HomeLayout />', () => {
  const clientOut = new ClientOutputModel();
    clientOut.firstName = "Eduardo"
    clientOut.lastName = "Pinedo"
    clientOut.cedula = "111111"

  test('Ui del componente', ()=> {
    const wraper = render(
      <ItemClientList item={clientOut} onPressEditClient={() => {}} />
    )
    expect(wraper.toJSON()).toMatchSnapshot()
  })

  test('Funcionalidad del onPress', ()=> {
    const onPress = jest.fn()
    const wraper = render(
      <ItemClientList item={clientOut} onPressEditClient={onPress} />
    )
    const buttonEdit = wraper.getByText('Editar')
    fireEvent.press(buttonEdit)

    expect(onPress).toHaveBeenCalledTimes(1)
  })
})