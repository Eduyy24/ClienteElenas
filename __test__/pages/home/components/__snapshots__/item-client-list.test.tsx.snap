// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemClientList /> Ui del componente 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 5,
    }
  }
>
  <View>
    <Text>
      Eduardo Pinedo
    </Text>
    <Text>
      111111
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Editar
    </Text>
  </View>
</View>
`;
